packages/ui/src/app/pages/Election/components/ElectionProgressBar.tsx

```diff
@@ -82,7 +82,10 @@ export const ElectionProgressBar = (props: ElectionProgressBarProps) => {
   const constants = useCouncilConstants()
 
   const [inactiveEndBlock, announcingEndBlock, votingEndBlock, revealingEndBlock] = periodInformation?.periodEnds ?? []
-  const endDates = periodInformation?.periodEnds.map((block) => blockToDate(block))
+  const endDates = useMemo(
+    () => periodInformation?.periodEnds.map((block) => blockToDate(block - currentBlock)),
+    [periodInformation?.currentStage]
```

Comment:
Good call! `periodInformation?.currentStage` is better since it will be defined when `currentBlock` will be defined.